{{ cookiecutter.repo_name }}/.gitignore

```diff
@@ -128,8 +122,600 @@ dmypy.json
 # Pyre type checker
 .pyre/
 
-# Pycharm
-.idea
+### dotenv ###
+.env
+
+### Flask ###
+instance/*
+!instance/.gitignore
+.webassets-cache
+
+### Flask.Python Stack ###
+# Byte-compiled / optimized / DLL files
+
+# C extensions
+
+# Distribution / packaging
+
+# PyInstaller
+#  Usually these files are written by a python script from a template
+#  before PyInstaller builds the exe, so as to inject date/other infos into it.
+
+# Installer logs
+
+# Unit test / coverage reports
+
+# Translations
+
+# Scrapy stuff:
+
+# Sphinx documentation
+
+# PyBuilder
+
+# pyenv
+
+# pipenv
+#   According to pypa/pipenv#598, it is recommended to include Pipfile.lock in version control.
+#   However, in case of collaboration, if having platform-specific dependencies or dependencies
+#   having no cross-platform support, pipenv may install dependencies that don't work, or not
+#   install all needed dependencies.
```

Comment:
Ah this I'm not sure! I copied and pasted the `.gitignore` from [gitignore.io](www.gitignore.io). Let me look into this.

Reply:
Okay - looking at the [issue](https://github.com/pypa/pipenv/issues/598) referenced in the comment.

`Pipfile.lock` is used to specify what package versions are used in a pipenv virtual environment ([reference](https://stackoverflow.com/a/49867443)).

The concern seems to involve collaboration - say two users are developing on the same repository using different `Pipfile.lock`s. In theory, both could have different packages (by version), which could cause problems overall. Hence the recommendation that it is always version controlled (which it is here!).

The issue suggests there could be some incompatibility issues for certain packages when using different operating systems. However, it these situations, it's more of an issue for the package maintainer to resolve, then the pipenv maintainers (see [here](https://github.com/pypa/pipenv/issues/598#issuecomment-377733910)). 

I think this is why there is a comment here around `Pipfile.lock`. I don't think there's a need to include it in the `requirements.txt` (I don't think this is how it works), just that, if there are os-specific issues, to uncomment out the `Pipfile.lock` line from `.gitignore` to essentially remove it from version control.

Reply:
Thanks Eric for this. 

I didn't mean to include `Pipfile.lock` in the `requirements.txt` but to extract the dependencies info in the former and save them as a `requirement.txt` to allow collaborators that do not use `pipenv` to recreate the development environment. But this is probably beyond the scope of this script.

[This discussion](https://stackoverflow.com/a/56060305/4014292) is also useful. It is recommended to synchronize the same `Pipfile.lock` with `pipenv sync` in order to share the exact same environment, simply version control it will not guarantee that.

Reply:
Ah okay, apologies my mistake. Not sure - but something worth considering for the future, if `pipenv` is used!